rosbag2_transport/src/rosbag2_transport/topic_filter.cpp

```diff
@@ -71,6 +71,12 @@ bool topic_in_list(const std::string & topic_name, const std::vector<std::string
   return it != topics.end();
 }
 
+bool type_in_list(const std::string & type_name, const std::vector<std::string> & topic_types)
```

Comment:
nit, just suggested name. in the future, we might need to add `service_type_in_list`, `action_type_in_list` as well.

```suggestion
bool topic_type_in_list(const std::string & type_name, const std::vector<std::string> & topic_types)
```